### DISCUSSION TITLE
Edit Schema

### DISCUSSION THREAD

**Discussion by nickap** [dgonz64/react-hook-form-auto#24] [General]
Hi there @dgonz64, 

I want to use this lib not only to render a form described by a schema but also to create a form from front end (form builder).
Is it feasible to have an empty schema in the first place, and to edit the schema after-wards?

I only see the  createSchema helper, while I would like to add or remove elements from schema afterwards.

Thanks!

**Comment by dgonz64** [comment]
Hello, nickap!

You can do something [like what the demo does](https://github.com/dgonz64/react-hook-form-auto-demo/blob/85133ad175bce35f5183d9bfdd2fcdff664c98d3/src/components/DemoForm.jsx#L66-L67), that is: create the schema every time there's a refresh. It's ugly and might lose form state, but it works already. You can try it [here](https://dgonz64.github.io/react-hook-form-auto-demo/demo/) by changing the schema definition.

The library is designed to have a fixed schema. The _correct_ implementation would have to make sure the new fields are registered and also unregister the deleted ones plus tests. Unfortunately I can't find time for it at the moment, but will leave the issue open just in case that changes or someone wants to chip in and write the code.
